### PR TITLE
feat(spans): Add more fields to span <-> transaction conversion

### PR DESCRIPTION
**Pull request by jjbayer**
Add some more fields to the span-transaction bidirectional mapping:

```rust
  span.data.segment_name <=> event.transaction, // transaction gets mapped to both segment_name and description
  span.data.browser_name <=> event.contexts.browser.name,
  span.data.sdk_name <=> event.client_sdk.name
```

This PR also fixes some problems with the `map_fields!` macro.

ref: #3278

#skip-changelog

### CODE REVIEW

File: relay-event-schema/src/protocol/span/convert.rs [hunk 1]
```diff
@@ -63,17 +66,20 @@ macro_rules! context_value (
...
     ($event:expr, contexts profile $context_field:ident) => {
         context_value!($event, ProfileContext, $context_field)
     };
-    ($event:expr, $path_root:ident $(. $path_segment:ident)*) => {
+    ($event:expr, $path_root:ident $($path_segment:ident)*) => {
         {
             let value = ($event).$path_root.value();
             $(
-                let value = value.and_then(|value|&value.$path_segment.value());
+                let value = value.and_then(|value|value.$path_segment.value());
```
Comment (jjbayer) [inline relay-event-schema/src/protocol/span/convert.rs:82]: Turned out I had some unused code branches here that needed to be fixed for deep access.

File: relay-event-schema/src/protocol/span.rs [hunk 2]
```diff
@@ -274,7 +274,8 @@ pub struct SpanData {
     /// Origin Transaction name of the span.
     ///
     /// For INP spans, this is the route name where the interaction occurred.
-    pub transaction: Annotated<String>,
+    #[metastructure(field = "segment.name", legacy_alias = "transaction")]
+    pub segment_name: Annotated<String>,
```
Comment (jjbayer) [inline relay-event-schema/src/protocol/span.rs:280]: Renamed to future key. `transaction` becomes a legacy alias.

Comment (iker-barriocanal) [reply relay-event-schema/src/protocol/span.rs:280]: Should we update the docs too?

File: relay-event-schema/src/protocol/span.rs [hunk 3]
```diff
@@ -292,6 +293,10 @@ pub struct SpanData {
     #[metastructure(field = "replay_id")]
     pub replay_id: Annotated<Value>,
 
+    #[metastructure(field = "sdk.name")]
+    /// The sentry SDK (see [`crate::protocol::ClientSdkInfo`]).
```
Comment (iker-barriocanal) [inline relay-event-schema/src/protocol/span.rs]:
```suggestion
    /// The sentry SDK (see [`crate::protocol::ClientSdkInfo`]).
    #[metastructure(field = "sdk.name")]
```

File: relay-event-schema/src/protocol/span/convert.rs [hunk 4]
```diff
@@ -310,6 +326,23 @@ mod tests {
         assert_eq!(event, roundtripped);
     }
 
+    #[test]
+    fn segment_name_takes_precedence() {
```
Comment (iker-barriocanal) [inline relay-event-schema/src/protocol/span/convert.rs]: What does it take precedence over?

**Review by iker-barriocanal — APPROVED** [review]
LGTM. It'd be great if we linked in the description the URL where the spec/mapping is defined.

File: relay-event-schema/src/protocol/span/convert.rs (hunk 4 above)
Comment (jjbayer) [reply relay-event-schema/src/protocol/span/convert.rs]: I updated the test name.

**Pull request merged**